src/main/java/org/zalando/nakadi/service/timeline/TimelineSyncImpl.java

```diff
@@ -0,0 +1,333 @@
+
+    @Scheduled(fixedDelay = 500)
+    public void reactOnEventTypesChange() throws InterruptedException {
+        checkForNewChange();
+        while (null != queuedChanges.peek()) {
+            final DelayedChange change = queuedChanges.peek();
+            LOG.info("Reacting on delayed change {}", change);
+            final Set<String> unlockedEventTypes = localLocking.lockedEventTypesChanged(change.lockedEventTypes);
+            // Notify consumers that they should refresh timeline information
+            for (final String unlocked : unlockedEventTypes) {
+                LOG.info("Notifying about unlock of {}", unlocked);
+                final List<Consumer<String>> toNotify;
+                synchronized (consumerListeners) {
+                    toNotify = consumerListeners.containsKey(unlocked) ?
+                            new ArrayList<>(consumerListeners.get(unlocked)) : null;
+                }
+                if (null != toNotify) {
+                    for (final Consumer<String> listener : toNotify) {
+                        try {
+                            listener.accept(unlocked);
+                        } catch (final RuntimeException ex) {
+                            LOG.error("Failed to notify about event type {} unlock", unlocked, ex);
+                        }
+                    }
+                }
+            }
+            try {
+                zooKeeperHolder.get().setData().forPath(
+                        toZkPath("/nodes/" + thisId), String.valueOf(change.version).getBytes(Charsets.UTF_8));
+            } catch (final Exception ex) {
+                LOG.error("Failed to update node version in zk. Will try to reprocess again", ex);
+                return;
+            }
+            queuedChanges.poll();
+            LOG.info("Delayed change {} successfully processed", change);
+        }
+    }
+
+    private void checkForNewChange() {
+        if (newVersionPresent.compareAndSet(true, false)) {
+            runLocked(() -> {
+                boolean success = false;
+                try {
+                    queuedChanges.add(
+                            new DelayedChange(readData("/version", this::versionChanged, Integer::parseInt),
+                                    this.zooKeeperHolder.get().getChildren().forPath(toZkPath("/locked_et"))));
+                    success = true;
+                } catch (final RuntimeException ex) {
+                    throw ex;
+                } catch (final InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                    throw new RuntimeException(ex);
+                } catch (final Exception ex) {
+                    throw new RuntimeException(ex);
+                } finally {
+                    if (!success) {
+                        newVersionPresent.set(true);
+                    }
+                }
+            });
+        }
+    }
+
+    private void versionChanged(final WatchedEvent ignore) {
```

Comment:
@adyach Well, yes. It is very good point. 
But here it doesn't matter what kind of change there is, because data will be re read and notification will be set up again.